05_multistep_td.py

```diff
@@ -172,14 +172,15 @@ def train(self):
             self.n_step_buffer.append(temp_transition)
             if len(self.n_step_buffer) == self.n_step:  # fill the n-step buffer for the first translation
                 # add a multi step transition
-                reward, next_obs, done = self.get_n_step_info(self.n_step_buffer, self.gamma)
+                reward, next_obs_to_store, done = self.get_n_step_info(self.n_step_buffer, self.gamma)
                 obs, action = self.n_step_buffer[0][:2]
-
+            else:
```

Comment:
can you make some explains about that improvement?

Reply:
It takes care of the cases where "done =True" is in the queue.
In that case the "next observation" is the "done state" and not the "real next observation".

The "obs, action = self.n_step_buffer[0][:2] : it is for the storage: You need to store the action and obs in time = t-n